stackslib/src/chainstate/stacks/boot/signers-voting.clar

```diff
@@ -0,0 +1,105 @@
+;;
+;; @contract voting for the aggregate public key
+;;
+
+;; maps dkg round and signer to proposed aggregate public key
+(define-map votes {reward-cycle: uint, round: uint, signer: principal} {aggregate-public-key: (buff 33), reward-slots: uint})
+;; maps dkg round and aggregate public key to weights of signers supporting this key so far
+(define-map tally {reward-cycle: uint, round: uint, aggregate-public-key: (buff 33)} uint)
+;; maps aggregate public keys to rewards cycles and rounds
+(define-map used-aggregate-public-keys (buff 33) {reward-cycle: uint, round: uint})
+
+(define-constant err-signer-index-mismatch (err u10000))
+(define-constant err-invalid-signer-index (err u10001))
+(define-constant err-out-of-voting-window (err u10002))
+(define-constant err-old-round (err u10003))
+(define-constant err-ill-formed-aggregate-public-key (err u10004))
+(define-constant err-duplicate-aggregate-public-key (err u10005))
+(define-constant err-duplicate-vote (err u10006))
+(define-constant err-invalid-burn-block-height (err u10007))
+
+(define-constant pox-info
+    (unwrap-panic (contract-call? .pox-4 get-pox-info)))
+
+;; maps reward-cycle ids to last round
+(define-map rounds uint uint)
+
+(define-data-var state-1 {reward-cycle: uint, round: uint, aggregate-public-key: (optional (buff 33)),
+    total-votes: uint}  {reward-cycle: u0, round: u0, aggregate-public-key: none, total-votes: u0})
+(define-data-var state-2 {reward-cycle: uint, round: uint, aggregate-public-key: (optional (buff 33)),
```

Comment:
Is state-2 for when we have hand off basically?

Reply:
Yes, when we alternate